### DISCUSSION TITLE
Consider renaming the hooks to avoid potential conflicts

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/evm-monorepo#595] [Ideas]
@razgraf has originally made this suggestion in https://github.com/sablier-labs/v2-periphery/discussions/112:

> The proxy plugin used for the Sablier on-cancel callback is called `onStreamCanceled`.
> 
> https://github.com/sablier-labs/v2-periphery/blob/453a35ef662183654138bfe8cab2b523f340fa5b/src/SablierV2ProxyPlugin.sol#L77-L79
> 
> Wouldn't renaming it in something more unique (e.g. `onSablierStreamCanceled`) help prevent unintended collisions in the future (e.g. other project using the PRBProxy doing "Potato Streaming")? That's not necessarily intended to improve security (as bad players could still engineer plugins with our original name) but unknowing developers may benefit from this.

However, I've opened another discussion here because it's the `ISablierV2LockupSender` hook that dictates what function the proxy plugin inherits:

https://github.com/sablier-labs/v2-core/blob/eb5014d7ef608da6a18aaa73ebd40d16dc899a88/src/interfaces/hooks/ISablierV2LockupSender.sol#L20-L26

Now, about the suggestion itself - it's a great idea, but we've unfortunately run out of time, and we won't be able to implement it if we want to stick to the July 10 deadline :(

Happy to re-consider the idea for V2.x

**Comment by PaulRBerg** [comment]
Any thoughts on this, @andreivladbrg, @smol-ninja?

We might be able to include this in V2.2

**Reply by andreivladbrg** [reply]
isn't it outdated since we've removed sender's hooks and the proxy arhitechture?

or do you say to also rename recipient's hooks?

```solidity
interface ISablierV2LockupRecipient {
     function onSablierStreamCanceled(uint256 streamId, address sender, uint128 senderAmount, uint128 recipientAmount) external;
     function onSablierStreamRenounced(uint256 streamId) external;
     function onSablierStreamWithdrawn(uint256 streamId, address caller, address to, uint128 amount) external;
}
```

for me it would be redundant as the interface name is `ISablierV2LockupRecipient` -  it already has the _Sablier_ aspect

**Reply by PaulRBerg** [reply]
> isn't it outdated since we've removed sender's hooks and the proxy arhitechture?

No. Razvan's proposal is about the names of all hooks in general, not about the sender specifically.

> do you say to also rename recipient's hooks?

Yes. Not me, Razvan said it. And I agree with his proposal.

> for me it would be redundant ...

Can you please re-read the original proposal?

**Reply by andreivladbrg** [reply]
i have and this is my take: 

> it would be redundant as the interface name is `ISablierV2LockupRecipient` - it already has the Sablier aspect

**Reply by PaulRBerg** [reply]
In this case, you've missed a crucial part of Razvan's point. Let me copy-paste it:

> Wouldn't renaming it in something more unique (e.g., `onSablierStreamCanceled`) help prevent unintended collisions in the future

The word "collisions" is key.

It's not about the name of the interface. It's about potential collisions with other interfaces. If we rename the hooks to include the name "Sablier", the problem is solved.

Do you understand now?

**Reply by andreivladbrg** [reply]
i do understand, but adding _Sablier_ is not good enough for me, i would rather suggest:

```solidity
interface ISablierV2Recipient {
     function onLockupStreamCanceled(uint256 streamId, address sender, uint128 senderAmount, uint128 recipientAmount) external;
     function onLockupStreamRenounced(uint256 streamId) external;
     function onLockupStreamWithdrawn(uint256 streamId, address caller, address to, uint128 amount) external;
     function onOpenEndedStreamCanceled(uint256 streamId, address sender, uint128 senderAmount, uint128 recipientAmount) external;
     function onOpenEndedStreamWithdrawn(uint256 streamId, address caller, address to, uint128 amount) external;
}
```

**Reply by smol-ninja** [reply]
I understand Razvan's argument. We should rename it if the motivation is developer experience.

**Reply by PaulRBerg** [reply]
> adding Sablier is not good enough for me, i would rather suggest:

This is a completely different idea, and I like it.

**Comment by andreivladbrg** [comment]
closing as we agreed, and created https://github.com/sablier-labs/v2-core/issues/736
